mailpoet/lib/Automation/Engine/Validation/WorkflowRules/ValidStepRule.php

```diff
@@ -45,9 +46,19 @@ public function visitNode(Workflow $workflow, WorkflowNode $node): void {
       try {
         $rule->visitNode($workflow, $node);
       } catch (UnexpectedValueException $e) {
-        $this->errors[$stepId] = ['step_id' => $stepId, 'message' => $e->getMessage()];
+        if (!isset($this->errors[$stepId])) {
+          $this->errors[$stepId] = ['step_id' => $stepId, 'message' => $e->getMessage(), 'fields' => []];
+        }
```

Comment:
This IF is to save the first error per field? Just asking if I understand.

Reply:
Yes.